src/App.tsx

```diff
@@ -1,18 +1,35 @@
-import React from 'react';
+import React, { ChangeEvent, useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [total] = useState(items.length);
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [itemsOnPage, setItemsOnPage] = useState([1, perPage]);
```

Comment:
add type for `itemsOnPage` state
probably would be better to create two separated states instead of one with array for two elements